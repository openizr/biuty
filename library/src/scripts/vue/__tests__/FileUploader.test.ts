/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { mount } from '@vue/test-utils';
import UIFileUploader from 'scripts/vue/FileUploader.vue';

type component = any; // eslint-disable-line @typescript-eslint/no-explicit-any

jest.mock('scripts/helpers/markdown');
jest.mock('scripts/helpers/generateRandomId');

describe('vue/UIFileUploader', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const wrapper = mount(UIFileUploader, {
      propsData: { name: 'test', modifiers: 'large' },
    });
    wrapper.find('input').trigger('focus');
    wrapper.find('input').trigger('change');
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    const wrapper = mount(UIFileUploader, {
      propsData: { name: 'test', id: 'test' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with placeholder', () => {
    const wrapper = mount(UIFileUploader, {
      propsData: { name: 'test', placeholder: 'text' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with accept', () => {
    const wrapper = mount(UIFileUploader, {
      propsData: { name: 'test', accept: 'image/*' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with helper', () => {
    const wrapper = mount(UIFileUploader, {
      propsData: { name: 'test', helper: 'Text' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with label', () => {
    const wrapper = mount(UIFileUploader, {
      propsData: { name: 'test', label: 'Text' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with left icon', () => {
    const wrapper = mount(UIFileUploader, {
      propsData: { name: 'test', icon: 'star' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with right icon', () => {
    const wrapper = mount(UIFileUploader, {
      propsData: { name: 'test', icon: 'star', iconPosition: 'right' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - multiple', () => {
    const wrapper = mount(UIFileUploader, {
      propsData: { name: 'test', multiple: true },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with listeners', async () => {
    const onFocus = jest.fn();
    const onChange = jest.fn();
    const wrapper = mount(UIFileUploader, {
      propsData: {
        name: 'test', icon: 'star', value: 'test',
      },
      listeners: {
        focus: onFocus,
        change: onChange,
      },
    });
    await (wrapper.vm as component).changeField({
      target: {
        files: [{ name: '/path/to/file1.png' }],
      },
    });
    await wrapper.vm.$nextTick();
    await (wrapper.vm as component).focusField();
    expect(wrapper.html()).toMatchSnapshot();
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith([{ name: '/path/to/file1.png' }]);
    expect(onFocus).toHaveBeenCalledTimes(1);
    expect(onFocus).toHaveBeenCalledWith(['/path/to/file1.png']);
  });
});
