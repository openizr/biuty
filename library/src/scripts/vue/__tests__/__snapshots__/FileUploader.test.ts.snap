// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue/UIFileUploader renders correctly - basic 1`] = `
"<div class=\\"ui-file-uploader ui-file-uploader--large\\">
  <!---->
  <!----> <input id=\\"z8a8d7a5zad4\\" type=\\"file\\" name=\\"test\\" class=\\"ui-file-uploader__field\\">
  <!----> <span class=\\"ui-file-uploader__files-list\\">

  </span>
  <!---->
</div>"
`;

exports[`vue/UIFileUploader renders correctly - multiple 1`] = `
"<div class=\\"ui-file-uploader\\">
  <!---->
  <!----> <input id=\\"z8a8d7a5zad4\\" type=\\"file\\" name=\\"test\\" multiple=\\"multiple\\" class=\\"ui-file-uploader__field\\">
  <!----> <span class=\\"ui-file-uploader__files-list\\">

  </span>
  <!---->
</div>"
`;

exports[`vue/UIFileUploader renders correctly - with helper 1`] = `
"<div class=\\"ui-file-uploader\\">
  <!---->
  <!----> <input id=\\"z8a8d7a5zad4\\" type=\\"file\\" name=\\"test\\" class=\\"ui-file-uploader__field\\">
  <!----> <span class=\\"ui-file-uploader__files-list\\">

  </span>
  <p class=\\"ui-file-uploader__helper\\">
    Text
  </p>
</div>"
`;

exports[`vue/UIFileUploader renders correctly - with id 1`] = `
"<div id=\\"test\\" class=\\"ui-file-uploader\\">
  <!---->
  <!----> <input id=\\"z8a8d7a5zad4\\" type=\\"file\\" name=\\"test\\" class=\\"ui-file-uploader__field\\">
  <!----> <span class=\\"ui-file-uploader__files-list\\">

  </span>
  <!---->
</div>"
`;

exports[`vue/UIFileUploader renders correctly - with label 1`] = `
"<div class=\\"ui-file-uploader\\"><label for=\\"z8a8d7a5zad4\\" class=\\"ui-file-uploader__label\\"><strong>Text</strong></label>
  <!----> <input id=\\"z8a8d7a5zad4\\" type=\\"file\\" name=\\"test\\" class=\\"ui-file-uploader__field\\">
  <!----> <span class=\\"ui-file-uploader__files-list\\">

  </span>
  <!---->
</div>"
`;

exports[`vue/UIFileUploader renders correctly - with left icon 1`] = `
"<div class=\\"ui-file-uploader\\">
  <!----> <i class=\\"ui-button__icon\\">star</i> <input id=\\"z8a8d7a5zad4\\" type=\\"file\\" name=\\"test\\" class=\\"ui-file-uploader__field\\">
  <!----> <span class=\\"ui-file-uploader__files-list\\">

  </span>
  <!---->
</div>"
`;

exports[`vue/UIFileUploader renders correctly - with listeners 1`] = `
"<div class=\\"ui-file-uploader\\" value=\\"test\\">
  <!----> <i class=\\"ui-button__icon\\">star</i> <input id=\\"z8a8d7a5zad4\\" type=\\"file\\" name=\\"test\\" class=\\"ui-file-uploader__field\\">
  <!----> <span class=\\"ui-file-uploader__files-list\\">
    /path/to/file1.png
  </span>
  <!---->
</div>"
`;

exports[`vue/UIFileUploader renders correctly - with placeholder 1`] = `
"<div class=\\"ui-file-uploader\\">
  <!---->
  <!----> <input id=\\"z8a8d7a5zad4\\" type=\\"file\\" name=\\"test\\" class=\\"ui-file-uploader__field\\">
  <!----> <span class=\\"ui-file-uploader__files-list\\">
    text
  </span>
  <!---->
</div>"
`;

exports[`vue/UIFileUploader renders correctly - with right icon 1`] = `
"<div class=\\"ui-file-uploader\\">
  <!---->
  <!----> <input id=\\"z8a8d7a5zad4\\" type=\\"file\\" name=\\"test\\" class=\\"ui-file-uploader__field\\"> <i class=\\"ui-button__icon\\">star</i> <span class=\\"ui-file-uploader__files-list\\">

  </span>
  <!---->
</div>"
`;
